### PR TITLE
8283041: [javadoc] Crashes using {@return} with @param

### PR DESCRIPTION
**Pull request by pavelrappo**
The inline `{@return}` tag is relatively new and will require developers to change their habits. According to the [specification](https://docs.oracle.com/en/java/javase/17/docs/specs/javadoc/doc-comment-spec.html#return), the inline version of `@return` "may only occur at the beginning of a method's description".

When used like in the description of the issue, the tag technically belongs to the block `@param` tag and not to the body of the doc comment, which one might think is the case. Thus, the "full body" (let alone "first sentence") collection of doc nodes is empty. Hence, IndexOutOfBoundsException when trying to access its first element. 

Since we don't have a method that returns the **complete** doc comment (yes, "getFullBody" is a bit of a misleading name), whose first element we could check against `{@return}`, I check `isEmpty()` before accessing the first element.

Interestingly, `{@summary}` (must also appear first) lint is performed differently. However, I decided not to copy it since it operates on a lower level of abstraction: characters and strings thereof.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283041](https://bugs.openjdk.java.net/browse/JDK-8283041): [javadoc] Crashes using {@return} with @param

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7788/head:pull/7788` \
`$ git checkout pull/7788`

Update a local copy of the PR: \
`$ git checkout pull/7788` \
`$ git pull https://git.openjdk.java.net/jdk pull/7788/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7788`

View PR using the GUI difftool: \
`$ git pr show -t 7788`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7788.diff">https://git.openjdk.java.net/jdk/pull/7788.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/Checker.java [hunk 1]
```diff
@@ -985,7 +985,7 @@ public Void visitReturn(ReturnTree tree, Void ignore) {
         }
         if (tree.isInline()) {
             DocCommentTree dct = getCurrentPath().getDocComment();
-            if (tree != dct.getFirstSentence().get(0)) {
+            if (dct.getFullBody().isEmpty() || tree != dct.getFullBody().get(0)) {
```
Comment (jonathan-gibbons) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/Checker.java:988]: It should not be necessary to resort to `getFullBody`.  It should be enough to check the first sentence, but that check should not throw an exception.

Comment (pavelrappo) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/Checker.java:988]: I thought that getting the first sentence would unnecessarily trigger sentence segmentation and read less clearly. But I can revert it if you like.

Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/Checker.java:988]: The split happens when the tree is created. See `DocTreeMaker` line 198.

**Comment by jonathan-gibbons** [comment]
As a general comment on terminology,

In the initial version of the API, a `DocCommentTree` was split into 3 parts:
1. the "first sentence"
2. the "body"
3. the block tags.

The split between "first sentence" and "body" eventually caused difficulties, because it lost the whitespace between them. So we retrofitted "full body" as the compound of "first sentence and body".

In a different reality, it would have been better to use the name `body" to refer to all the content before the block tags, and to have some other name for the content between the first sentence and the block tags, which (with hindsight) is typically a less interesting abstraction.   The rationale is that we did not want "overlap" between nodes of the tree.

**Review by jonathan-gibbons — APPROVED** [review]

**Pull request closed without merging**